### PR TITLE
fix mypy crash on empty class parse error

### PR DESCRIPTION
**Pull request by graingert**
```
class : 

class Ham:
    def __init__(self) -> None: ...
```

### CODE REVIEW

**Comment by graingert** [comment]
```pytb
mypy foo.py
Traceback (most recent call last):
  File "/home/graingert/.virtualenvs/mypy/bin/mypy", line 6, in <module>
    main(__file__)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/main.py", line 54, in main
    res = type_check_only(sources, bin_dir, options)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/main.py", line 102, in type_check_only
    python_path=options.python_path)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 209, in build
    dispatch(sources, manager)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1253, in dispatch
    graph = load_graph(sources, manager)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1268, in load_graph
    st = State(id=bs.module, path=bs.path, source=bs.text, manager=manager)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1040, in __init__
    self.parse_file()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1151, in parse_file
    self.tree = manager.parse_file(self.id, self.xpath, source)
  File "/usr/lib/python3.5/contextlib.py", line 77, in __exit__
    self.gen.throw(type, value, traceback)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1105, in wrap_context
    yield
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 1151, in parse_file
    self.tree = manager.parse_file(self.id, self.xpath, source)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/build.py", line 448, in parse_file
    fast_parser=FAST_PARSER in self.flags)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 96, in parse
    tree = parser.parse(source)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 147, in parse
    file = self.parse_file()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 174, in parse_file
    defs = self.parse_defs()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 320, in parse_defs
    defn, is_simple = self.parse_statement()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 951, in parse_statement
    stmt = self.parse_expression_or_assignment()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 957, in parse_expression_or_assignment
    expr = self.parse_expression(star_expr_allowed=True)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 1327, in parse_expression
    self.parse_error()
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 1861, in parse_error
    self.parse_error_at(self.current())
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/parse.py", line 1875, in parse_error_at
    self.errors.report(tok.line, msg)
  File "/home/graingert/.virtualenvs/mypy/lib/python3.5/site-packages/mypy/errors.py", line 162, in report
    type = self.type_name[-1]
IndexError: list index out of range

*** INTERNAL ERROR ***

foo.py: error: Internal error -- please report a bug at https://github.com/python/mypy/issues
```

File: mypy/errors.py [hunk 1]
```diff
@@ -159,7 +159,12 @@ def report(self, line: int, message: str, blocker: bool = False,
             file: if non-None, override current file as context
             only_once: if True, only report this exact message once per build
         """
-        type = self.type_name[-1]
```
Comment (graingert) [inline mypy/errors.py]: could be something like '<no name>'

**Comment by gvanrossum** [comment]
I propose not to fix this. The new parser will reject this invalid syntax, and the hack in errors.py doesn't look very elegant.

**Comment by graingert** [comment]
@gvanrossum is there somewhere I can use this new parser?

**Comment by gvanrossum** [comment]
Yes, use `--new-parser`. You also have to `pip3 install typed-ast`.

**Comment by graingert** [comment]
`git grep -i "new.parser"` shows nothing...

**Comment by gvanrossum** [comment]
Sorry, it's `--fast-parser`.

**Comment by graingert** [comment]
@gvanrossum yeah that works.

**Pull request closed without merging**